fbw-a32nx/src/systems/systems-host/systems/FWC/PseudoFWC.ts

```diff
@@ -1332,6 +1501,184 @@ export class PseudoFWC {
+
+    this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.write(
+      this.autoPilotInstinctiveDiscPressedPulse.read() &&
+        (this.autoPilotInstinctiveDiscCountSinceLastFwsCycle > 1 || apDiscPressedInLast1p8SecBeforeThisCycle),
+      deltaTime,
+    );
+
+    this.autoPilotOffVoluntaryMemory.write(
+      this.autoPilotOffVoluntaryDiscPulse.read(),
+      apEngaged ||
+        this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() ||
+        !this.autoPilotOffVoluntaryEndAfter1p9s.read(),
+    );
+
+    const discPbPressedAfterDisconnection =
+      !this.autoPilotDisengagedInstantPulse.read() &&
+      (this.autoPilotInstinctiveDiscPressedPulse.read() || masterWarningButtonLeft || masterWarningButtonRight);
+
+    this.autoPilotOffUnacknowledged.write(
+      this.autoPilotDisengagedInstantPulse.read(),
+      apEngaged || this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() || discPbPressedAfterDisconnection,
+    );
+
+    this.autoPilotOffInvoluntaryMemory.write(
+      !apEngaged && !this.autoPilotOffVoluntaryMemory.read() && this.autoPilotOffUnacknowledged.read(),
+      !this.autoPilotOffUnacknowledged.read(),
+    );
+    this.autoPilotOffInvoluntary.set(this.autoPilotOffInvoluntaryMemory.read());
+    this.autoPilotOffShowMemo.set(this.autoPilotOffVoluntaryMemory.read() || this.autoPilotOffInvoluntaryMemory.read());
+
+    if (this.autoPilotDisengagedInstantPulse.read()) {
+      // Request quiet CRC one time
+      this.requestMasterWarningFromApOff = true;
+      this.soundManager.enqueueSound('cavalryChargeOnce'); // On the A320, play first cav charge completely no matter what
+    }
+    if (!this.autoPilotOffVoluntaryFirstCavalryChargeActive.read()) {
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+    }
+    if (!this.autoPilotOffVoluntaryMemory.read() && !this.autoPilotOffInvoluntaryMemory.read()) {
+      this.requestMasterWarningFromApOff = false;
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+      this.soundManager.dequeueSound('cavalryChargeCont');
+    }
+
+    this.autoPilotInstinctiveDiscPressedPulse.write(false, deltaTime);
+
+    // approach capability downgrade. Debounce first, then suppress for a certain amount of time
+    // (to avoid multiple triple clicks, and a delay which is too long)
+    let fmgcApproachCapability = 0;
+    const getApproachCapability = (dw4: Arinc429WordData): number => {
+      let appCap = 0;
+      const landModeActive = dw4.bitValueOr(14, false);
+      const land2Capacity = dw4.bitValueOr(23, false);
+      const land3FailPassiveCapacity = dw4.bitValueOr(24, false);
+      const land3FailOperationalCapacity = dw4.bitValueOr(25, false);
+
+      if (land2Capacity) {
+        appCap = 2;
+      } else if (land3FailPassiveCapacity) {
+        appCap = 3;
+      } else if (land3FailOperationalCapacity) {
+        appCap = 4;
+      } else if (landModeActive) {
```

Comment:
The check for land mode armed is  missing which could explain why the triple click doesn't play with approach mode armed.

Reply:
added now, thanks for looking into it!